bug(frontend): App.vue has 3 pre-existing TS2322 errors — SVG fill-rule and Booleanish type mismatches

## Discovery Source
Found during implementation of #4546 — agent noted pre-existing errors in App.vue unchanged by the fix.

## Problem
`autobot-frontend/src/App.vue` has 3 TypeScript errors:

```
src/App.vue(57,52):  error TS2322: Type 'string | undefined' is not assignable to type '"inherit" | "evenodd" | "nonzero" | undefined'.
src/App.vue(113,16): error TS2322: Type 'string' is not assignable to type 'Booleanish | undefined'.
src/App.vue(159,48): error TS2322: Type 'string | undefined' is not assignable to type '"inherit" | "evenodd" | "nonzero" | undefined'.
```

Lines 57 and 159: SVG `fill-rule` attribute bound to a string variable, but Vue's SVG type expects the literal union `"inherit" | "evenodd" | "nonzero" | undefined`.
Line 113: A prop typed as `string` passed where `Booleanish | undefined` is expected.

## Fix
- Lines 57, 159: Cast or constrain the variable to the SVG fill-rule literal union, e.g. `:fill-rule="(svgVar as 'inherit' | 'evenodd' | 'nonzero' | undefined)"`
- Line 113: Pass `true`/`false` boolean literal or cast to `Booleanish`

## Files
- `autobot-frontend/src/App.vue` lines 57, 113, 159

--- a/autobot-frontend/src/App.vue
+++ b/autobot-frontend/src/App.vue
@@ -110,7 +110,7 @@
               @click="toggleMobileNav"
               class="lg:hidden inline-flex items-center justify-center p-2 rounded text-autobot-text-primary hover:bg-autobot-bg-tertiary focus:outline-none focus:ring-2 focus:ring-autobot-primary"
               aria-controls="mobile-nav"
-              :aria-expanded="showMobileNav.toString()"
+              :aria-expanded="showMobileNav"
             >
               <span class="sr-only">{{ $t('nav.openMainMenu') }}</span>
               <!-- Hamburger / X icon toggle (#1804) -->
@@ -763,7 +763,17 @@ export default {
     const slmAdminUrl = computed(() => getSLMAdminUrl());
 
     // Data-driven navigation items: single source of truth for desktop + mobile nav
-    const navItems = [
+    // iconRule is typed as a literal union to satisfy SVG fill-rule / clip-rule prop types (#4699)
+    type SvgFillRule = 'evenodd' | 'nonzero' | 'inherit';
+    const navItems: Array<{
+      to: string;
+      labelKey: string;
+      icon?: string;
+      iconPaths?: string[];
+      iconRule?: SvgFillRule;
+      iconStroke?: boolean;
+      adminOnly?: boolean;
+    }> = [
       { to: '/home', labelKey: 'nav.home', icon: 'M10.707 2.293a1 1 0 00-1.414 0l-7 7v11a1 1 0 001 1h2a1 1 0 001-1v-5a1 1 0 011-1h2a1 1 0 011 1v5a1 1 0 001 1h2a1 1 0 001-1v-7l7-7a1 1 0 000-1.414z', iconRule: 'evenodd' },
       { to: '/about', labelKey: 'nav.about', icon: 'M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z', iconStroke: true },
       { to: '/chat', labelKey: 'nav.chat', icon: 'M18 10c0 3.866-3.582 7-8 7a8.841 8.841 0 01-4.083-.98L2 17l1.338-3.123C2.493 12.767 2 11.434 2 10c0-3.866 3.582-7 8-7s8 3.134 8 7zM7 9H5v2h2V9zm8 0h-2v2h2V9zM9 9h2v2H9V9z', iconRule: 'evenodd' },

